### PR TITLE
Rewrite IngestArchivalMediaBagJob and related classes

### LINKED ISSUE #3020: Archival media bag ingest should accommodate `_pt1`, etc.
In several cases the vendor had to split one side of a recording into multiple parts.

example:
https://github.com/pulibrary/figgy/issues/3018

### PR DESCRIPTION
**Pull request by tpendragon**
Implement new hierarchical model structure as per #3021
closes #3020
Generally clean up and refactor IngestArchivalMediaBagJob and
ArchivalMediaBagParser

We tried this out, and it ingests correctly, but there's no way to get from the descriptive proxy to the MediaResource with the FileSets. We expect we'll fix this as part of converting to using Recordings (#3080)

### CODE REVIEW

**Comment by hackmastera** [comment]
To the reviewer: For IngestArchivalMediaBagJob and ArchivalMediaBagParser this is enough of a rewrite that the diffs might not be super helpful and you may want to look at the files directly.

File: spec/fixtures/files/pulfa/C0652/c0383.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
-               Collections</corpname><subarea>Manuscripts Division</subarea><address><addressline>One Washington Road</addressline><addressline>Princeton, New Jersey 08544 USA</addressline></address></repository><container type="item" parent="C0652_i22">8</container><physdesc><extent type="computed" unit="folders">1 folder</extent></physdesc></did><altformavail><p>32101047382484_1_a.mp3</p><p>32101047382484_2_a.mp3</p><p>32101047382492_2_a.mp3</p></altformavail><scopecontent><p>2 audiocassettes; Tape 1, Side A written description on tape: Fitas / Monegal
+               Collections</corpname><subarea>Manuscripts Division</subarea><address><addressline>One Washington Road</addressline><addressline>Princeton, New Jersey 08544 USA</addressline></address></repository><container type="item" parent="C0652_i22">8</container><physdesc><extent type="computed" unit="folders">1 folder</extent></physdesc></did><altformavail><p>32101047382492_2_a.mp3</p><p>32101047382484_1_a.mp3</p><p>32101047382484_2_a.mp3</p></altformavail><scopecontent><p>2 audiocassettes; Tape 1, Side A written description on tape: Fitas / Monegal
```
Comment (escowles) [inline spec/fixtures/files/pulfa/C0652/c0383.xml]: Are there any real changes to these fixture files? One change is an added linebreak, but I can't see what the change is for the other two lines.

Comment (tpendragon) [reply spec/fixtures/files/pulfa/C0652/c0383.xml]: Yeah, it's hard to see. The order of the altformavails are switched to make the ordering test fail.

Comment (hackmastera) [reply spec/fixtures/files/pulfa/C0652/c0383.xml]: We had to switch the order of the barcodes to adequately test the order of the resulting resource members in one of our tests; we needed to manipulate the order in the file vs. the filesystem sort order.

Comment (escowles) [reply spec/fixtures/files/pulfa/C0652/c0383.xml]: 👍 OK, that makes sense and lines up with the test that uses it

**Comment by escowles** [comment]
@hackmastera Also, thanks for the prompt to go look at the files instead of the diff — that was indeed much easier to read.

**Pull request merged**